datasets/fr/senat/crawler.py

```diff
@@ -75,16 +80,38 @@ def crawl_row(context: Context, row: Dict[str, str]):
         context.emit(occupancy)
 
 
+def crawl_mandates(
+    context: Context, reader: Iterator[Dict[str, str]]
+) -> Iterator[Tuple[str, str]]:
+    """Get start dates for senators by ID."""
+    for row in reader:
+        # Skip former senators
+        if row.pop("État Sénateur") != "ACTIF":
+            continue
+        # Skip reelection events
+        if row.pop("Motif début de mandat") != "Election":
+            continue
+        senid = row.pop("Matricule")
+        dt = row.pop("Date de début de mandat")
+        start_date = dt.split(maxsplit=1)[0]
+        yield senid, start_date
```

Comment:
so we actually like getting a detailed picture of someone's tenure - creating an occupancy entity for each mandate with its own start and end dates. We also like including ex senators who are still considered PEPs based on mandate end date by `h.make_occupancy()`

for sources where we can assume we want all the entities from both files or API listing endpoints and we can generate the same ID from both files, we often just create a person entity in both - they'll get deduped at the export stage.

In cases like this, where we know there could be ex-senators in the senator file who would be excluded because their last national gov mandate ended more than 20 years ago, I tend to keep a set of person IDs who were considered PEPs, and then use that when crawling the senator detail file to decide who to include (rather than using `État Sénateur`)

Reply:
Good to know!  It makes sense since the data often seem to be structured that way.

The senator file has *all* the senators since the start of the 5th republic so we can add them all.  By "were considered PEPs" do you mean the categorization given by `zavod.logic.pep.categorize`?  In this case should I call `categorize` without `is_pep=True`?

Reply:
nope, the categorize call is fine

`make_occupancy` has some logic that decides whether the occupancy implies that the person is a PEP based on a bunch of rules including when they departed the position https://zavod.opensanctions.org/peps/#creating-occupancies